examples/Flock.jl

```diff
@@ -0,0 +1,130 @@
+# # Flock model
+
+# The flock model illustrates how flocking behavior can emerge when each bird follows three simple rules:
+#
+# * maintain a minimum distance from other birds to avoid collision
+# * fly towards the average position of neighbors
+# * fly in the average direction of neighbors
+
+
+# ## Defining the core structures
+
+# We begin by calling the required packages and defining an agent type representing a bird.
+
+using Agents, Random, LinearAlgebra
+
+mutable struct Bird <: AbstractAgent
+    id::Int
+    pos::NTuple{2,Float64}
+    vel::NTuple{2,Float64}
+    speed::Float64
+    cohere_factor::Float64
+    separation::Float64
+    seperate_factor::Float64
+    match_factor::Float64
+    visual_distance::Float64
+end
+
+# The fields `id` and `pos` are required for every agent. The field `vel` is required for every agent in ContinuousSpace.
+# `speed` defines how quickly the bird travels in a direction defined by `vel`. `seperation` defines the minimum distance between a bird must maintain from its neighbors.
+# `visual_distance` refers to the distance a bird can see and defines a radius of neighboring birds.
+# The contribution of each rule defined above recieves an importance weight: `cohere_factor` is the importance of maintaining the average position of neighbors,
+# `match_factor` is the importance of matching the average trajectory of neighboring birds, and `seperate_factor` is the importance of maining the minimum
+# distance from neighboring birds.
+#
+# The following defines a Bird constructor that uses keyword arguments and default values.
+function Bird(id, pos; vel=(0.0,0.0), speed=1.0, cohere_factor=.25, separation=4.0,
+        seperate_factor=.25, match_factor=.01, visual_distance=5.0)
+    return Bird(id, pos, vel, speed, cohere_factor, separation, seperate_factor,
+        match_factor, visual_distance)
+end
```

Comment:
Why do this? Is there any benefit? It makes the code bigger. In the following `initialize_model` function, the fields are known.

Reply:
kwargs are useful if you want to generate objects with default values and modify default values without remembering the positions. I removed that in the latest commit.